### PR TITLE
[libharu] Fix HPDF_DLL export symbol

### PR DESCRIPTION
**Pull request by Meakk**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- ~~[ ] SHA512s are updated for each updated download~~
- ~~[ ] The "supports" clause reflects platforms that may be fixed by this new version~~
- ~~[ ] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Does this change actually fix a port problem? The header fixup might still be desired for those who do not use CMake.

**Comment by Meakk** [comment]
The variable `HDF_DLL` should not be defined on Linux when building shared libs, otherwise ther are compilation issues for ports consuming libharu.  
I can revert most of the changes and add a condition to check if we are on Windows before patching the headers.

**Comment by Meakk** [comment]
@dg0yt let me know if it looks better now.

File: ports/libharu/portfile.cmake [hunk 1]
```diff
@@ -13,17 +13,14 @@ vcpkg_from_github(
...
   set(VCPKG_BUILD_SHARED_LIBS OFF)
-  set(VCPKG_BUILD_STATIC_LIBS ON)
 endif()
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
-        -DLIBHPDF_STATIC=${VCPKG_BUILD_STATIC_LIBS}
-        -DLIBHPDF_SHARED=${VCPKG_BUILD_SHARED_LIBS}
+        -DBUILD_SHARED_LIBS=${VCPKG_BUILD_SHARED_LIBS}
```
Comment (dg0yt) [inline ports/libharu/portfile.cmake:23]: This variables is already controlled by `vcpkg_cmake_configure`. It seems that libharu no longer needs custom code for `VCPKG_LIBRARY_LINKAGE`.

File: ports/libharu/portfile.cmake [hunk 2]
```diff
@@ -40,17 +38,16 @@ file(REMOVE_RECURSE
...
     string(REPLACE "#ifdef HPDF_DLL\n" "#if 0\n" _contents "${_contents}")
 endif()
 file(WRITE "${CURRENT_PACKAGES_DIR}/include/hpdf.h" "${_contents}")
 
 file(READ "${CURRENT_PACKAGES_DIR}/include/hpdf_types.h" _contents)
-if(VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
+if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
     string(REPLACE "#ifdef HPDF_DLL\n" "#if 1\n" _contents "${_contents}")
 else()
     string(REPLACE "#ifdef HPDF_DLL\n" "#if 0\n" _contents "${_contents}")
```
Comment (dg0yt) [inline ports/libharu/portfile.cmake:53]: I would simplify the whole block to:
~~~cmake
if(VCPKG_TARGET_IS_WINDOWS AND VCPKG_LIBRARY_LINKAGE STREQUAL "dynamic")
    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/hpdf.h" "#ifdef HPDF_DLL\n" "#if 1\n")
    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/hpdf_types.h" "#ifdef HPDF_DLL\n" "#if 1\n")
endif()
~~~
(Untested.)

Comment (Meakk) [reply ports/libharu/portfile.cmake]: Tested on Windows and Linux locally, seems to work fine. It looks much nicer, thanks for the suggestion.

**Comment by Thomas1664** [comment]
Note: The problem you're fixing here (i.e. setting some DLL-related variables on non-Windows) seems to be a common issue. I searched for `if\(VCPKG_LIBRARY_LINKAGE STREQUAL "(static|dynamic)"\)(\n|\r\n).*DLL` in all portfiles and found about 30 hits. In most cases, this won't cause any issues, but this PR shows that we have to be careful with this.
Note: there are false positive ports included by the search string that only support Windows or check for Windows in a previous if-statement.

**Comment by Meakk** [comment]
Let me take a look at why it's failing on static x64 and x86 triplets.

**Comment by Meakk** [comment]
I'm a bit confused, I do not reproduce the failing builds with `x64-windows-static` and `x86-windows`.

**Comment by jimwang118** [comment]
Thanks for your PR, Is work still being done for this PR?

**Comment by Meakk** [comment]
> Thanks for your PR, Is work still being done for this PR?

Sorry I have no time to finish this PR, I'm closing it.

**Pull request closed without merging**